File: frontend/lib/seller/storeCategories.ts
export type SellerStoreCategoryOption = {
  label: string;
  slug: string;
  query: string;
  iconKey: string | null;
};

export const SELLER_STORE_CATEGORY_OPTIONS: SellerStoreCategoryOption[] = [
  { label: "مد و پوشاک", slug: "fashion", query: "fashion clothing apparel", iconKey: "shirt" },
  { label: "زیبایی", slug: "beauty", query: "beauty skincare makeup", iconKey: "sparkles" },
  { label: "الکترونیک", slug: "electronics", query: "electronics gadgets tech", iconKey: "cpu" },
  { label: "خانه و دکور", slug: "home", query: "home decor household", iconKey: "home" },
  { label: "دیجیتال", slug: "digital", query: "digital services downloads", iconKey: "download" },
  { label: "اکسسوری", slug: "accessories", query: "accessories gifts add-ons", iconKey: "box" },
  { label: "سایر", slug: "other", query: "other", iconKey: null },
];

export function findSellerStoreCategory(slug: string | null | undefined): SellerStoreCategoryOption | undefined {
  if (!slug) return undefined;
  return SELLER_STORE_CATEGORY_OPTIONS.find((category) => category.slug === slug.trim().toLowerCase());
}

File: backend/app/core/store_categories.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class StoreCategoryOption:
    label: str
    slug: str
    query: str
    icon_key: str | None
    keywords: tuple[str, ...]


STORE_CATEGORIES: tuple[StoreCategoryOption, ...] = (
    StoreCategoryOption(
        "مد و پوشاک",
        "fashion",
        "fashion clothing apparel",
        "shirt",
        ("shirt", "dress", "hoodie", "jacket", "polo", "fashion", "clothes"),
    ),
    StoreCategoryOption(
        "زیبایی",
        "beauty",
        "beauty skincare makeup",
        "sparkles",
        ("beauty", "makeup", "skincare", "cosmetic", "serum", "cream"),
    ),
    StoreCategoryOption(
        "الکترونیک",
        "electronics",
        "electronics gadgets tech",
        "cpu",
        ("phone", "laptop", "tablet", "earbud", "charger", "tech", "electronics"),
    ),
    StoreCategoryOption(
        "خانه و دکور",
        "home",
        "home decor household",
        "home",
        ("home", "kitchen", "decor", "household", "furniture", "room"),
    ),
    StoreCategoryOption(
        "دیجیتال",
        "digital",
        "digital services downloads",
        "download",
        ("digital", "ebook", "course", "template", "download", "service"),
    ),
    StoreCategoryOption(
        "اکسسوری",
        "accessories",
        "accessories gifts add-ons",
        "box",
        ("accessory", "bag", "case", "wallet", "gift", "accessories"),
    ),
)

OTHER_CATEGORY = StoreCategoryOption("سایر", "other", "other", None, ())


def list_store_categories() -> list[StoreCategoryOption]:
    return [*STORE_CATEGORIES, OTHER_CATEGORY]


def get_store_category(slug: str | None) -> StoreCategoryOption | None:
    if not slug:
        return None

    normalized = slug.strip().lower()
    for category in STORE_CATEGORIES:
        if category.slug == normalized:
            return category
    if normalized == OTHER_CATEGORY.slug:
        return OTHER_CATEGORY
    return None
